### PR TITLE
Emit same id in ctxt json as in bfrt json

### PR DESCRIPTION
**Pull request by kamleshbhalui**
Emit same handle id in context json as in p4info and bfrt.

### CODE REVIEW

**Comment by kamleshbhalui** [comment]
context json:

{
  "program_name" : "psa-example-dpdk-byte-alignment_1",
  "build_date" : "Sun Jan  8 21:43:34 2023",
  "compile_command" : "./p4c-dpdk /home/kumarkam/workarea4/p4c/testdata/p4_16_samples/psa-example-dpdk-byte-alignment_1.p4 -o kk.spec --arch psa --context foo.json --p4runtime-files foox.json -v",
  "compiler_version" : "0.1 (SHA: b0a9a7536 BUILD: Debug)",
  "schema_version" : "0.1",
  "target" : "DPDK",
  "tables" : [
    {
      "name" : "ingress.tbl",
      "target_name" : "ingress.tbl",
      "direction" : "ingress",
      "handle" : 44506256,
      "table_type" : "match",
      "size" : 65536,
      "p4_hidden" : false,
      "add_on_miss" : false,
      "idle_timeout_with_auto_delete" : false,
      "stateful_table_refs" : [],
      "statistics_table_refs" : [],
      "meter_table_refs" : [],
      "match_key_fields" : [
        {
          "name" : "hdr.ethernet.srcAddr",
          "instance_name" : "hdr.ethernet",
          "field_name" : "srcAddr",
          "match_type" : "exact",
          "start_bit" : 0,
          "bit_width" : 48,
          "bit_width_full" : 48,
          "position" : 0
        }
      ],
      "actions" : [
        {
          "name" : "NoAction",
          "target_name" : "NoAction",
          "handle" : 21257015,
          "constant_default_action" : false,
          "is_compiler_added_action" : false,
          "allowed_as_hit_action" : true,
          "allowed_as_default_action" : true,
          "p4_parameters" : []
        },
        {
          "name" : "ingress.execute",
          "target_name" : "ingress.execute_1",
          "handle" : 29480552,
          "constant_default_action" : false,
          "is_compiler_added_action" : false,
          "allowed_as_hit_action" : true,
          "allowed_as_default_action" : true,
          "p4_parameters" : [
            {
              "name" : "index",
              "start_bit" : 0,
              "bit_width" : 12,
              "position" : 0,
              "byte_array_index" : 0
            }
          ]
        }
      ],
      "match_attributes" : {
        "stage_tables" : [
          {
            "action_format" : [
              {
                "action_name" : "NoAction",
                "action_handle" : 21257015,
                "immediate_fields" : []
              },
              {
                "action_name" : "ingress.execute",
                "action_handle" : 29480552,
                "immediate_fields" : [
                  {
                    "param_name" : "index",
                    "dest_start" : 0,
                    "dest_width" : 12
                  }
                ]
              }
            ]
          }
        ]
      },
      "default_action_handle" : 21257015
    }
  ],
  "externs" : [
    {
      "name" : "ingress.counter0",
      "target_name" : "counter0_0",
      "type" : "Counter",
      "attributes" : {
        "type" : "packets_and_bytes"
      }
    },
    {
      "name" : "ingress.counter1",
      "target_name" : "counter1_0",
      "type" : "Counter",
      "attributes" : {
        "type" : "packets"
      }
    },
    {
      "name" : "ingress.counter2",
      "target_name" : "counter2_0",
      "type" : "Counter",
      "attributes" : {
        "type" : "bytes"
      }
    },
    {
      "name" : "ingress.reg",
      "target_name" : "reg_0",
      "type" : "Register",
      "attributes" : {
      }
    },
    {
      "name" : "ingress.meter0",
      "target_name" : "meter0_0",
      "type" : "Meter",
      "attributes" : {
      }
    }
  ]

**Comment by kamleshbhalui** [comment]
p4info:

{
 "pkgInfo": {
  "arch": "psa"
 },
 "tables": [
  {
   "preamble": {
    "id": 44506256,
    "name": "ingress.tbl",
    "alias": "tbl"
   },
   "matchFields": [
    {
     "id": 1,
     "name": "hdr.ethernet.srcAddr",
     "bitwidth": 48,
     "matchType": "EXACT"
    }
   ],
   "actionRefs": [
    {
     "id": 21257015
    },
    {
     "id": 29480552
    }
   ],
   "size": "1024"
  }
 ],
 "actions": [
  {
   "preamble": {
    "id": 21257015,
    "name": "NoAction",
    "alias": "NoAction",
    "annotations": [
     "@noWarn(\"unused\")"
    ]
   }
  },
  {
   "preamble": {
    "id": 29480552,
    "name": "ingress.execute",
    "alias": "execute"
   },
   "params": [
    {
     "id": 1,
     "name": "index",
     "bitwidth": 12
    }
   ]
  },
  {
   "preamble": {
    "id": 30967736,
    "name": "ingress.test",
    "alias": "test"
   }
  }
 ],
 "counters": [
  {
   "preamble": {
    "id": 314114784,
    "name": "ingress.counter0",
    "alias": "counter0"
   },
   "spec": {
    "unit": "BOTH"
   },
   "size": "1024"
  },
  {
   "preamble": {
    "id": 318473483,
    "name": "ingress.counter1",
    "alias": "counter1"
   },
   "spec": {
    "unit": "PACKETS"
   },
   "size": "1024"
  },
  {
   "preamble": {
    "id": 313911423,
    "name": "ingress.counter2",
    "alias": "counter2"
   },
   "spec": {
    "unit": "BYTES"
   },
   "size": "1024"
  }
 ],
 "meters": [
  {
   "preamble": {
    "id": 344514043,
    "name": "ingress.meter0",
    "alias": "meter0"
   },
   "spec": {
    "unit": "BYTES"
   },
   "size": "1024"
  }
 ],
 "registers": [
  {
   "preamble": {
    "id": 382856063,
    "name": "ingress.reg",
    "alias": "reg"
   },
   "typeSpec": {
    "bitstring": {
     "bit": {
      "bitwidth": 32
     }
    }
   },
   "size": 1024
  }
 ],
 "typeInfo": {}
}

File: backends/dpdk/dpdkContext.cpp [hunk 1]
```diff
@@ -200,6 +188,33 @@ Util::JsonObject* DpdkContextGenerator::initTableCommonJson(const cstring name,
...
+
+size_t DpdkContextGenerator::getHandleId(cstring name) {
+    size_t id = 0;
+    for (auto x : context_handle_map) {
+        if (x.first.find(name.c_str())) {
+            id = context_handle_map[x.first];
+            break;
+        }
+    }
+    return id;
```
Comment (psivanup) [inline backends/dpdk/dpdkContext.cpp:215]: Is it ok to return 0 if handle is not found? 
IMO, either fallback to older method or asserting would be fine.

Comment (usha1830) [reply backends/dpdk/dpdkContext.cpp:215]: Yes I think asserting would be better.

Comment (kamleshbhalui) [reply backends/dpdk/dpdkContext.cpp:216]: added bugcheck

**Review by psivanup — APPROVED** [review]
Looks fine to me.

**Review by syogender — APPROVED** [review]

**Pull request closed without merging**